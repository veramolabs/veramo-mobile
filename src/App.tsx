import React, { useEffect, useState } from 'react'
import {
  SafeAreaView,
  ScrollView,
  View,
  Text,
  Button,
  TextInput,
} from 'react-native'

// Import agent from setup
import { agent } from './veramo'

interface Identifier {
  did: string
  alias?: string
}

const App = () => {
  const [identifiers, setIdentifiers] = useState<Identifier[]>([])
  const [alias, setAlias] = useState<string>('')

  // Add the new identifier to state
  const createIdentifier = async () => {
    try {
      const _id = await agent.didManagerCreate({ alias })
      setIdentifiers((s) => s.concat([_id]))
      setAlias('')
    } catch (error) {
      console.log(error)
    }
  }

  // Check for existing identifers on load and set them to state
  useEffect(() => {
    const getIdentifiers = async () => {
      const _ids = await agent.didManagerFind()
      setIdentifiers(_ids)

      // Inspect the id object in your debug tool
      // console.log('_ids:', _ids)
    }

    getIdentifiers()
  }, [])

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <ScrollView testID="scrollView">
        <View>
          <Text style={{ fontSize: 30, fontWeight: 'bold', padding: 20 }}>
            Identifiers
          </Text>
          <View
            style={{
              marginBottom: 50,
              marginTop: 20,
              margin: 15,
              borderRadius: 20,
              backgroundColor: '#EFEFEF',
            }}
          >
            {identifiers && identifiers.length > 0 ? (
              identifiers.map((id: Identifier, index: number) => (
                <View
                  key={id.did}
                  style={{
                    padding: 15,
                    borderBottomColor: '#DDDDDD',
                    borderBottomWidth: identifiers.length > index + 1 ? 1 : 0,
                  }}
                >
                  <Text
                    style={{ fontSize: 18, fontWeight: 'bold' }}
                    testID={'item-identifier'}
                  >
                    {id.alias}
                  </Text>
                  <Text>{id.did}</Text>
                </View>
              ))
            ) : (
              <Text>No identifiers created yet</Text>
            )}
          </View>
          <View style={{ padding: 15 }}>
            <TextInput
              testID="input"
              placeholder="Add alias name (required)"
              style={{
                marginBottom: 20,
                backgroundColor: '#EFEFEF',
                padding: 15,
                borderRadius: 10,
              }}
              onChangeText={(text) => setAlias(text)}
              value={alias}
            />
            <Text>{alias}</Text>
            <Button
              disabled={!alias}
              onPress={() => createIdentifier()}
              title={'Create Identifier'}
            />
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  )
}

export default App
