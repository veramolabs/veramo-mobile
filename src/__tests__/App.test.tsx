// import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'
import App from '../App'

describe('App', () => {
  it('does not explode', async () => {
    const { toJSON } = render(<App />)

    expect(toJSON()).toMatchSnapshot()
  })
})
