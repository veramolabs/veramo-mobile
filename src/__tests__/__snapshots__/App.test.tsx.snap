// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App does not explode 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
        >
          Identifiers
        </Text>
        <View
          style={
            Object {
              "marginBottom": 50,
              "marginTop": 20,
            }
          }
        >
          <Text>
            No identifiers created yet
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Create Identifier
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
